ui/src/pages/deployment-details/deployment-details-form/section-location/geo-search/useGeoSearch.ts

```diff
@@ -0,0 +1,57 @@
+import axios from 'axios'
+import { MarkerPosition } from 'design-system/map/types'
+import _ from 'lodash'
+import { useCallback, useEffect, useState } from 'react'
+import { SearchResult, ServerSearchResult } from './types'
+
+const API_URL = '/nominatim'
+const MAX_NUM_RESULTS = 5
+
+const convertServerResult = (result: ServerSearchResult): SearchResult => {
+  const latitude = _.toNumber(result.lat)
+  const longitude = _.toNumber(result.lon)
+
+  return {
+    osmId: result.osm_id,
+    displayName: result.display_name,
+    position: new MarkerPosition(latitude, longitude),
+  }
+}
+
+const getFetchUrl = (searchString: string) =>
+  `${API_URL}?format=json&q=${searchString}`
+
+export const useGeoSearch = (searchString: string) => {
+  const [fetchUrl, setFetchUrl] = useState<string>()
+  const debouncedSetFetchUrl = useCallback(_.debounce(setFetchUrl, 200), [
```

Comment:
If typing fast we don't want to call the API on each change

Reply:
I wonder if it would be cleaner to keep this method "debounce free", and then debounce the usage of useGeoSearch?

Reply:
Yes good point, that got cleaner! I could not debounce the usage of useGeoSearch (rules of hooks), but I extracted the debouncing from it. What do you think?

Reply:
Looks good!